From d6d2dfca261a46c66efecbb297b0a997d347f0d2 Mon Sep 17 00:00:00 2001
From: Hector Cao <hector.cao@canonical.com>
Date: Mon, 23 Jun 2025 13:08:19 +0000
Subject: [PATCH] ubuntu sauce
Last-Update: 2025-08-22
---
--- intel-dpcpp-6.2.0.orig/opencl/CMakeLists.txt
+++ intel-dpcpp-6.2.0/opencl/CMakeLists.txt
@@ -23,29 +23,12 @@ set(OCL_LOADER_REPO
 set(OCL_HEADERS_TAG 542d7a8f65ecfd88b38de35d8b10aa67b36b33b2)
 set(OCL_LOADER_TAG 804b6f040503c47148bee535230070da6b857ae4)
 
-# OpenCL Headers
-if(NOT OpenCL_HEADERS)
-  message(STATUS "Will fetch OpenCL headers from ${OCL_HEADERS_REPO}")
-
-  FetchContent_Declare(ocl-headers
-    GIT_REPOSITORY    ${OCL_HEADERS_REPO}
-    GIT_TAG           ${OCL_HEADERS_TAG}
-  )
-else()
-  message(STATUS "OpenCL headers are added manually ${OpenCL_HEADERS}")
-
-  FetchContent_Declare(ocl-headers
-    URL               ${OpenCL_HEADERS}
-  )
-endif()
-
-FetchContent_MakeAvailable(ocl-headers)
-FetchContent_GetProperties(ocl-headers)
+find_package(OpenCLHeaders REQUIRED)
 set(OpenCL_INCLUDE_DIR
-  ${ocl-headers_SOURCE_DIR} CACHE PATH "Path to OpenCL Headers")
+  ${OpenCL_INCLUDE_DIR} CACHE PATH "Path to OpenCL Headers")
 
-target_compile_definitions(Headers INTERFACE -DCL_TARGET_OPENCL_VERSION=300)
-add_library(OpenCL-Headers ALIAS Headers)
+target_compile_definitions(OpenCL::Headers INTERFACE -DCL_TARGET_OPENCL_VERSION=300)
+add_library(OpenCL-Headers ALIAS OpenCL::Headers)
 
 # OpenCL Library (ICD Loader)
 
@@ -57,25 +40,7 @@ set(OPENCL_ICD_LOADER_HEADERS_DIR
 # library.
 set(BUILD_SHARED_LIBS ON)
 
-if(NOT OpenCL_LIBRARY_SRC)
-  message(STATUS "Will fetch OpenCL ICD Loader from ${OCL_LOADER_REPO}")
-
-  FetchContent_Declare(ocl-icd
-    GIT_REPOSITORY    ${OCL_LOADER_REPO}
-    GIT_TAG           ${OCL_LOADER_TAG}
-  )
-else()
-  # TODO: add possibility to use prebuilt OpenCL library rather than building
-  #       together with llvm.
-  message(STATUS
-    "OpenCL ICD Loader sources added manually ${OpenCL_LIBRARY_SRC}")
-
-  FetchContent_Declare(ocl-icd
-    URL               ${OpenCL_LIBRARY_SRC}
-  )
-endif()
-
-FetchContent_MakeAvailable(ocl-icd)
-add_library(OpenCL-ICD ALIAS OpenCL)
+find_package(OpenCL QUIET)
+add_library(OpenCL-ICD ALIAS OpenCL::OpenCL)
 
 add_subdirectory(opencl-aot)
--- intel-dpcpp-6.2.0.orig/sycl/cmake/modules/FetchUnifiedRuntime.cmake
+++ intel-dpcpp-6.2.0/sycl/cmake/modules/FetchUnifiedRuntime.cmake
@@ -428,8 +428,3 @@ if(CMAKE_SYSTEM_NAME STREQUAL Windows)
       DESTINATION "bin" COMPONENT unified-memory-framework)
   endif()
 endif()
-
-install(TARGETS umf
-  LIBRARY DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT unified-memory-framework
-  ARCHIVE DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT unified-memory-framework
-  RUNTIME DESTINATION "bin" COMPONENT unified-memory-framework)
--- intel-dpcpp-6.2.0.orig/sycl/tools/sycl-ls/CMakeLists.txt
+++ intel-dpcpp-6.2.0/sycl/tools/sycl-ls/CMakeLists.txt
@@ -14,7 +14,6 @@ target_compile_definitions(sycl-ls PRIVA
 target_link_libraries(sycl-ls
   PRIVATE
     ${sycl_lib}
-    OpenCL-Headers
     UnifiedRuntime-Headers
 )
 if (WIN32)
