From 6b6857c07162d1723cdc355cfb8dfe2a79659f92 Mon Sep 17 00:00:00 2001
From: Hector Cao <hector.cao@canonical.com>
Date: Wed, 25 Jun 2025 21:50:41 +0000
Subject: [PATCH] does not copy level zero loader headers around
Last-Update: 2025-08-22
---
--- intel-dpcpp-6.2.0.orig/unified-runtime/cmake/FetchLevelZero.cmake
+++ intel-dpcpp-6.2.0/unified-runtime/cmake/FetchLevelZero.cmake
@@ -98,8 +98,8 @@ target_link_libraries(LevelZeroLoader
     INTERFACE "${LEVEL_ZERO_LIB_NAME}"
 )
 
-file(GLOB LEVEL_ZERO_LOADER_API_HEADERS "${LEVEL_ZERO_INCLUDE_DIR}/*.h")
-file(COPY ${LEVEL_ZERO_LOADER_API_HEADERS} DESTINATION ${LEVEL_ZERO_INCLUDE_DIR}/level_zero)
+# file(GLOB LEVEL_ZERO_LOADER_API_HEADERS "${LEVEL_ZERO_INCLUDE_DIR}/*.h")
+# file(COPY ${LEVEL_ZERO_LOADER_API_HEADERS} DESTINATION ${LEVEL_ZERO_INCLUDE_DIR}/level_zero)
 add_library(LevelZeroLoader-Headers INTERFACE)
 target_include_directories(LevelZeroLoader-Headers
     INTERFACE "$<BUILD_INTERFACE:${LEVEL_ZERO_INCLUDE_DIR};${LEVEL_ZERO_INCLUDE_DIR}/level_zero>"
